.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,46 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
```

Comment:
Was a particular reason you created a separate file instead of adding this GH action the the `.github/workflows/pr.yml` file? Was it because this job is somehow incompatible with the others, or was it to keep things organized?

Reply:
mostly to keep things organized!  I can add it to the PR check if that would make more sense!